### ISSUE TITLE
[Feature Request] Silent sh execution

### ISSUE THREAD

**Issue by wolfcharm** [ForkIssues/TrackerWin#1844]
For custom commands, if sh command executed, the console window popups for a half second. This behaviour seems unpleasant. Can we get some sort of checkbox for silent execution?

**Comment by DanPristupov** [comment]
> For custom commands, if sh command executed, the console window popups for a half second. 

How is your custom command configured?

I don't see any window myself.

<img width="643" alt="Screenshot 2023-03-20 at 14 31 08" src="https://user-images.githubusercontent.com/618115/226354953-c36a4996-8814-423b-8fda-b7dc6cbb715e.png">

<img width="491" alt="Screenshot 2023-03-20 at 14 31 21" src="https://user-images.githubusercontent.com/618115/226355016-cf9a8c7a-ff8e-46e3-8035-339f9dc5752d.png">

**Comment by wolfcharm** [comment]
> > For custom commands, if sh command executed, the console window popups for a half second.
> 
> How is your custom command configured?
> 
> I don't see any window myself.

![image](https://user-images.githubusercontent.com/12260557/226619660-da40286d-1980-4203-b41f-87d9ab74cffe.png)
![image](https://user-images.githubusercontent.com/12260557/226619760-98c02cdc-593e-4656-afc6-68bbf26de866.png)

**Comment by DanPristupov** [comment]
We made a few improvements in 1.83. Could you check if the problem is solver?

**Comment by Texl** [comment]
I think the improvements in 1.83 may be the cause of an issue I'm seeing now. I've been using custom command shell scripts to run git-p4 - for example:
```bash
branch=$(git symbolic-ref --short HEAD)
echo [git-p4] rebase branch $branch
git p4 rebase
read -p 'Press any key to continue . . . ' -s -n 1
```
I've been running these with "Wait for Exit" turned off because git-p4 is unable to run with it turned on - it requires access to stdin and isn't able to get it:
```
Traceback (most recent call last):
...
  File "C:\Python39\lib\subprocess.py", line 1252, in _get_handles
    p2cread = _winapi.GetStdHandle(_winapi.STD_INPUT_HANDLE)
OSError: [WinError 6] The handle is invalid
```
This was fine with me - with "Wait for Exit" turned off, commands ran successfully in a console window. I even added a bit of interactivity in some scripts since I had access to stdin:
```bash
echo [git-p4] update-shelve commit $sha
read -p 'Enter shelved changelist number: ' cl
git p4 submit --update-shelve $cl --commit $SHA
read -p 'Press any key to continue . . . ' -s -n 1
```

With the new changes in 1.83, the console window no longer opens - so I'd say you were successful :) My commands are probably just running silently now, but unfortunately git-p4 is too unreliable with my p4 server for me to be able to trust that it's succeeding. (It requires a lot of manual intervention, the p4 server I'm working with has no standards for text encoding.) I've tried it with "Wait for Exit" checked too after the update, but it continues to have those stdin issues.

What do you recommend I do to get commands like the one above working again? Do you expect shell script custom commands to work this way at all in the future, or should I just write a small program to handle them and invoke it via Start Process custom commands?

**Comment by DanPristupov** [comment]
@Texl yes, this is a problem. Yes, we need to have a way to run GUI applications. I will think about that.

**Comment by DanPristupov** [comment]
@Texl 

`.sh` actions are supposed to be silent. Fork doesn't handle std-input itself and I'm surprised you found a way to use it. I wasn't aware it's possible :). In other words, that was a bug in Fork for Windows, which you used as a feature.

Right now I can not make a fix which would 'just work' for everyone (especially on both OSs).

So, what you should probably do is to start `sh.exe` as a GUI application and pass the script into it.

How to do that:

1. Create custom command with `Start Process` action, ensure that `Wait for exit` is disabled. This way Fork will start process as a GUI app.

2. Create .sh script `git-p4-rebase.sh` with the following content:

```
#!/bin/sh

branch=$(git symbolic-ref --short HEAD)
echo [git-p4] rebase branch $branch
git p4 rebase
read -p 'Press any key to continue . . . ' -s -n 1
```

3. Pass the file path as an argument to `sh`.

<img width="592" alt="Screenshot 2023-04-28 at 12 23 22" src="https://user-images.githubusercontent.com/618115/235122807-407cf46f-a423-46f7-84c2-938254af4bf3.png">

~~(I noticed that relative paths are not supported, but I'll fix that in the 1.84 update.)~~
Both absolute and relative paths are supported. So, you can also use a relative path to script and store it in the repository, if needed.

**Comment by Texl** [comment]
Sounds good. Thank you for the guidance - I figured it was best to ask for your advice directly in order to make sure I'm using the feature as expected.

Thanks for a great product - I recommend it to everyone who asks :)
